docs/proposals/2017-04-20 - Charts.md

```diff
@@ -0,0 +1,201 @@
+
+The following proposal deals specifically with the use case of creating a basic data chart.
+
+# Proposal
+
+A new `chart` format that includes row/column settings and a citation, and models cells as text elements (rather than strings.)
+
+## Title
+
+An ANS `text` element.
+
+## Citation
+
+An ANS `text` element, identical to a `citation` on the new `quote` element.  Intended to indicate the source of the data.
+
+## Credits
+
+An ANS credits object, can denormalize authors from author service. Intended to indicate the person who created this specific chart.
+
+## No width/height/sizing options
+
+Explicit sizing options are deliberately not included.
+
+## Explicitly Defined Columns - `columns`
+
+Data columns should be explicitly defined in the `columns` attribute. The following attributes can be set:
+
+* `_id` Automatically-generated.
+* `text_alignment` can be optionally set and applies to the entire column.
+* `title` text element can be optionally supplied here which can function as a header for a column.  **If at least one column has a title than a header row containing the title should be added to the chart. Otherwise it should be omitted.**
+* `borders` an array of strings with the possible values `"left"` and `"right"`
+
+## Explicitly Defined Rows - `rows`
+
+* `_id` Automatically-generated.
+* `title` text element can be optionally supplied here which can function as a header for a row.  **If at least one row has a title then a header column containing the title should be added to the chart. Otherwise it should be omitted.**
+* `borders` an array of strings with the possible values `"top"` and `"bottom"`
+* `cells` an array of cell objects.
+
+## Cells
+
+Currently these objects have just two properties:
+
+* `_id` - Automatically-generated
+* `cell_content` - currently this object can only be an ANS `text` element.
+
+# Example
+
+```
+{
+  "type": "story",
+  "version": "0.5.9",
+  "content_elements": [
+    {
+      "type": "chart",
+
+      "title": {
+        "type": "text",
+        "content": "My Awesome Chart"
+      },
+
+      "citation": {
+        "type": "text",
+        "content": "Source: The Washington Post"
+      },
+
+      "credits": {
+        "by": [{
+          "type": "author",
+          "name": "Angela Wong"
+        }]
+      },
+
+      "columns": [
+        {
+          "_id": "AAAAAAAAAAAAAAAAAAAAAA",
+          "text_alignment": "left",
+          "borders": [ "right" ],
+
+          "title": {
+            "type": "text",
+            "content": "Men"
+          }
+        },
+        {
+          "_id": "AAAAAAAAAAAAAAAAAAAAAB",
+          "text_alignment": "right",
+
+          "title": {
+            "type": "text",
+            "content": "Women"
+          }
+
+        }
+      ],
+
```

Comment:
You're right, in the HTML you absolutely need to use colspan/rowspan, but you don't need them in the ANS to add them in the HTML.  The ANS object has four rows and two columns. In an HTML `<table>`, this example might be rendered as five rows and three columns -- the extra header rows/columns being inferred from the presence of `title`.  (That's what I was alluding to with "If at least one row has a title than a header column containing the title should be added to the chart. Otherwise it should be omitted." That way the renderer can also decide to change the formatting on the header rows/columns to differentiate them, or use another solution entirely)